### ISSUE TITLE
Selection of items from `SelectMultiple` is broken after content update via observe.

### ISSUE THREAD

**Issue by mlovci** [ipython/ipywidgets#643]
My goal is to have two widgets, `select` and `other` that accept a list of options.

`select` may only take on one value, so I chose the `RadioButtons` widget for this. `other` may take on multiple values, so I used `SelectMultiple`. Further, the options available in `other` may not contain the value chosen in `select.` Ideally (but optionally), the selected items in `other` should be cleared when a new item is chosen from `select`. 

I used `widget.observe` to update `other` when `select` changes, however (here's the bug) the selection area in the `SelectMultiple` does not respond correctly to clicks after updates. For example, when I click one value, another becomes highlighted, or when I select a range an alternate set of values becomes selected. The behavior is erratic and inconsistent, so you might have to try a few times to get it to misbehave.

This code updates the content of `other` when the selection of `select` is altered:

```
from ipywidgets import SelectMultiple, RadioButtons, interact
options = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h']

select_widget = RadioButtons(options=options)
other_widget = SelectMultiple(options=sorted(list(set(select_widget.options).difference(set([select_widget.value])))),
                              selected_labels=())
other_widget.selected_labels = ()

def update_other(*args):

    other_widget.value = ()
    other_widget.selected_labels = ()
    other_widget.options = list(set(select_widget.options).difference(set([select_widget.value])))

select_widget.observe(update_other)

def printer(select, other):
    print(select, other)

interact(printer, select=select_widget, other=other_widget, __manual = True)
```

Thank you,
Mike

**Comment by PhML** [comment]
I obtain the same behaviour with this code:

```python
from IPython.display import display
import ipywidgets as ipw

parents = ['pa', 'pb', 'pc', 'pd']
children = {'pa': 1, 'pb': 2, 'pc': 3, 'pd': 4}

parents_selector = ipw.SelectMultiple(description="Sites", options=parents)

select_all_parents = ipw.Button(description='Select all parents')
def on_select_all_parents_clicked(b):
    parents_selector.value = parents_selector.options
select_all_parents.on_click(on_select_all_parents_clicked)

def on_selected_parents_change(change):
    children_selector.options = list(children[child] for child in change['new'])
parents_selector.observe(on_selected_parents_change, 'value')

children_selector = ipw.SelectMultiple(description="children")
def on_children_selector_change(change):
    print(change)
children_selector.observe(on_children_selector_change, 'value')


display(parents_selector, select_all_parents, children_selector)
```

**Comment by gnestor** [comment]
@mlovci I just tried out your sample code using ipywidgets at master and it works as expected. Try upgrading to the latest version of ipywidgets (5.2.2) and let me know if you're still having issues...

**Comment by PhML** [comment]
@gnestor Have you tried my sample?
First select some sites, then a child, then some other sites. Now, when you try to select a child, it often «jump» to another one than the one that you selected.
I use the version 5.2.2.

**Comment by jasongrout** [comment]
I just tried both samples with current master (6.0.0b2 + some commits), and they seem to work fine.

I did notice another bug, which I fixed: #883.

**Comment by gnestor** [comment]
I can verify that it's working on master. I'm closing. Re-open if you see this in the next release.
